src/Microsoft.DotNet.DeadRegionAnalysis/AnalysisEngine.Options.cs

```diff
@@ -0,0 +1,200 @@
+                IEnumerable<string> projectPaths = null,
+                IEnumerable<string> sourcePaths = null,
+                IEnumerable<IEnumerable<string>> symbolConfigurations = null,
+                IEnumerable<string> alwaysIgnoredSymbols = null,
+                IEnumerable<string> alwaysDefinedSymbols = null,
+                IEnumerable<string> alwaysDisabledSymbols = null,
+                Tristate undefinedSymbolValue = default(Tristate))
+            {
+                if (projectPaths != null)
+                {
+                    projects = Task.WhenAll(from path in projectPaths select MSBuildWorkspace.Create().OpenProjectAsync(path, CancellationToken.None)).Result;
+                }
+                if (projects != null)
+                {
+                    Documents = GetSharedDocuments(projects);
+                }
+
+                if (sourcePaths != null)
+                {
+                    var projectId = ProjectId.CreateNewId("AnalysisProject");
+                    var solution = new AdhocWorkspace()
+                        .CurrentSolution
+                        .AddProject(projectId, "AnalysisProject", "AnalysisProject", LanguageNames.CSharp);
+
+                    foreach (var path in sourcePaths)
+                    {
+                        var documentId = DocumentId.CreateNewId(projectId);
+                        solution = solution.AddDocument(
+                            documentId,
+                            Path.GetFileName(path),
+                            new FileTextLoader(path, Encoding.UTF8));
+                    }
+
+                    Documents = solution.Projects.Single().Documents;
+                }
+
+                _symbolConfigurations = CalculateSymbolConfigurations(
+                    alwaysDisabledSymbols,
+                    alwaysDefinedSymbols,
+                    alwaysIgnoredSymbols,
+                    symbolConfigurations);
+
+                _undefinedSymbolValue = undefinedSymbolValue;
+            }
+
+            internal CompositePreprocessorExpressionEvaluator GetPreprocessorExpressionEvaluator()
+            {
+                var evaluators = _symbolConfigurations.Select(config => new PreprocessorExpressionEvaluator(config, _undefinedSymbolValue));
+                return new CompositePreprocessorExpressionEvaluator(evaluators);
+            }
+
+            internal PreprocessorSymbolTracker GetPreprocessorSymbolTracker()
+            {
+                var specifiedSymbols = new HashSet<string>();
+
+                foreach (var config in _symbolConfigurations)
+                {
+                    foreach (string symbol in config.Keys)
+                    {
+                        specifiedSymbols.Add(symbol);
+                    }
+                }
+
+                return new PreprocessorSymbolTracker(specifiedSymbols);
+            }
+
+            private static IEnumerable<Document> GetSharedDocuments(IEnumerable<Project> projects)
```

Comment:
Correction: I thought I had changed this, but it turns out I am not ignoring the symbol configuration specified in the project files in the AnalysisEngine.FromProjects() method (currently only used by tests).  I am currently using it in addition to any configurations specified with the `/config` option.  Given my explanation above though, I think it makes sense never to use the symbol configuration deduced by simply loading the project file in the current environment.